Extend build with transport

aradi opened the pull request:
Also use the customized source archive instead of GitHubs
automatically generated one to ensure that the RELEASE file is
contained.

Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

aradi commented:
@conda-forge-admin please rerender

aradi committed:
Extend build with transport

Also use the customized source archive instead of GitHubs
automatically generated one to ensure that the RELEASE file is
contained.

aradi commented:
@conda-forge-admin please rerender

Commit:
MNT: Re-rendered with conda-build 3.21.4, conda-smithy 3.8.6, and conda-forge-pinning 2021.01.31.00.18.44

awvwgk commented:
CMake's detection for MPI is currently broken in mpich 3.4.x (https://github.com/conda-forge/mpich-feedstock/issues/56).

Pinning mpich to not 3.4.x in host seems to work for now, but is quite ugly:

```yaml
  host:
    - {{ mpi }}  # [mpi != "nompi"]
    - mpich !=3.4  # [mpi == "mpich"]
```

I would recommend to wait for upstream to fix this issue instead of messing with the host dependencies.

aradi commented:
@awvwgk Do you have by any chance clue, why the OSX with MPICH fails? I can not see, why any of my changes should have caused this failure... (I guess, OSX and me are not really compatible for some reasons :wink: )

aradi commented:
Makes sense, the PR is not too urgent.

awvwgk commented:
@conda-forge-admin, please restart CI

awvwgk commented:
@aradi what is the status on this patch? The CI is passing with the MPICH fix so we could move this forward now.

awvwgk commented on recipe/meta.yaml:
```diff
@@ -42,6 +42,7 @@ requirements:
     - mpifx * mpi_{{ mpi }}_*  # [mpi != "nompi"]
     - scalapack  # [mpi != "nompi"]
     - scalapackfx * mpi_{{ mpi }}_*  # [mpi != "nompi"]
+    - libnegf * {{ mpi_prefix }}_*
```
The correct `libnegf` version is picked automatically for non-MPI builds, constraints only apply for MPI builds:
```suggestion
    - libnegf
    - libnegf * mpi_{{ mpi }}_*  # [mpi != "nompi"]
```

awvwgk reviewed (changes requested):
Also `libnegf` should be pinned in the run dependencies correctly.

awvwgk commented:
@aradi What is the status on this PR?

awvwgk committed:
Add libnegf as run dependency

MNT: Re-rendered with conda-build 3.21.4, conda-smithy 3.9.0, and conda-forge-pinning 2021.02.26.15.36.03

awvwgk committed:
Pin compatibile negf version
